### ISSUE TITLE
Handle empty changelog after a release from an older branch

### ISSUE THREAD

**Issue by bajtos** [strongloop/strong-tools#22]
Today, I made three releases of loopback-boot:

 - [v2.6.2](https://github.com/strongloop/loopback-boot/releases/tag/v2.6.2) publishing one commit
 - [v1.1.2](https://github.com/strongloop/loopback-boot/releases/tag/v1.1.2) publishing a back-port of the commit above
 - [v2.6.3](https://github.com/strongloop/loopback-boot/releases/tag/v2.6.3) to fix the README on npmjs.org

The release `v1.1.2` was made via `npm release patch && npm publish`, both `v2.6.2` and `v2.6.3` were made using `slt-release patch`.

**Actual behaviour**

The changelog generated for `v2.6.3` seems to contain all commits from the master after 1.x branch was forked.

**Expected behaviour**

The changelog generated for `v2.6.3` is empty.

/to @sam-github @rmg do you happen to know where to start investigating this problem?

**Comment by rmg** [comment]
Multiple/parallel releases (maintaining 2.x and 1.x "production" branches) is something not well supported by these tools.

For the specific problem you are seeing, it is likely the result of the release tags being listed in chronological order and the the changelog entries for each release being relative to the previous release _in that order_. Because all release tags are merged into production and then into master, changing the order to graph order would not help any.

Basically we need to support multiple base/production branches to fix your problem.

**Comment by bajtos** [comment]
>  Because all release tags are merged into production and then into master, changing the order to graph order would not help any

I believe this is not true. Since the 1.x release was not produced by the tool, the release tag was not merged into master nor into production. Changing the order to graph order IMO should solve the problem in my particular case.

**Comment by rmg** [comment]
I meant `slt-release` will merge the release back in to master. But you're right. If the branches aren't merged it should work. Another thing we can do is add a reachability filter to the tag listing, so that only releases reachable from the current commit are considered. It would result in independently maintained change logs for each maintenance release branch, but I think that would be better than the current situation.

**Comment by bajtos** [comment]
> Add a reachability filter to the tag listing, so that only releases reachable from the current commit are considered. It would result in independently maintained change logs for each maintenance release branch, but I think that would be better than the current situation.

That sounds like a perfect solution to me.

**Comment by rmg** [comment]
Released as v2.0.0 on npmjs.org
